frontend/templates/product-list/sections/ProductListChildrenSection.tsx

```diff
@@ -33,9 +38,21 @@ export function ProductListChildrenSection({
       [isShowingMore]
    );
 
+   const { items: deviceMenuItems } = useMenu({
+      attribute: 'device',
+      limit: PRODUCT_LIST_MAX_FACET_VALUES_COUNT,
+   });
```

Comment:
I think it's better if we use our `useMenuFacet` here to make sure that the Algolia widget gets created with the same parameters everywhere so we avoid weird behaviors

Reply:
I had considered that, but doing so would display at most 10 empty children links (which defeats the purpose of the PR), since `limit=10` in `useMenuFacet` for the `device` facet.

This implementation intersects the device facet items with the children links to determine which children links are empty, so we want all possible `device` facet items that are applicable to the current collection. The assumption is that if the `device` facet item is not present, then there are no products in the collection that work in that device, and so we should exclude the device from the child link section.

However, now I realize this assumption does not hold if other facets are selected, since the list of applicable device facets will dwindle as the collection is filtered.

![image](https://user-images.githubusercontent.com/52104630/205101332-a0d8c7fc-cccc-4ef7-838e-b7878f908008.png)

Reply:
@sterlinghirsh It doesn't seem like Algolia offers a way to ask "Give me all the facet item values (i.e. device titles) for the unfiltered state of the current collection", meaning that this solution might not work.

Reply:
Seems we have two options. One is that we do a second query for the full unfiltered list. The other is that we accept that other filters may have an effect on the model list. I'm okay with the latter as long as the UI doesn't jump around too much.

Reply:
We should probably do this filtering server side only, in the product list model so we're not affecting in any possible way client side filtering

Reply:
> One is that we do a second query for the full unfiltered list

To keep this request separate from the instantsearch state, we could use `fetch` or install algolia's [JS api client](https://www.algolia.com/doc/api-client/getting-started/install/javascript/?client=javascript) to get the list of device titles on the server.

> The other is that we accept that other filters may have an effect on the model list.

This feels strange given that we clear all filters when users click on a child device link.

> as long as the UI doesn't jump around too much

The layout shift doesn't seem like great UX.

[Screencast from 12-02-2022 10:43:50 AM.webm](https://user-images.githubusercontent.com/52104630/205330992-7f1f92d3-d5dd-4cb3-8ccd-d8f90ceed765.webm)

Reply:
One option is that when we remove child productlist options that were previously visible, we gray them out and sort to the bottom instead of hiding them. Not quite as clean, but it reduces the layout shift.

Reply:
> One option is that when we remove child productlist options that were previously visible, we gray them out and sort to the bottom instead of hiding them. Not quite as clean, but it reduces the layout shift.

@sterlinghirsh I tried this suggestion in https://github.com/iFixit/react-commerce/pull/1091/commits/43f3bba03d48519237ecd975d23ee37d651be639. I think the UI looks good, but it appears there are sometimes inconsistencies between the device titles on Algolia and the device titles on Strapi.

For example, on https://react-commerce-git-no-empty-child-collections-ifixit.vercel.app/Parts/Electronics "VR Gear" is greyed out, but it has plenty of products when you click on it. I'm not sure if it's just out of date data in Algolia, or an actual discrepancy between the two.

Reply:
I see that too, though I think this is a data issue. On electronics parts, searching for vive doesn't return all the vive parts. I think on cominor Virtual Reality Headset is not a child of Electronics or something.

Reply:
But on the live site it appears to work.